wagtail/query.py

```diff
@@ -521,7 +521,7 @@ def annotate_has_untranslated_locale(self):
         )
 
 
-class SpecificIterable(BaseIterable):
+class SpecificIterable(ModelIterable):
```

Comment:
Do you think we should change this, or should we raise a ticket to Django asking them to check against `ValuesIterable`, `ValuesListIterable`, and `FlatValuesListIterable` instead of requiring a `ModelIterable` for `in_bulk` (https://github.com/django/django/pull/18496)?

Reply:
I can see an argument for requesting Django loosen its checks for what classes of objects should be allowed to use `in_bulk`. But if I understand Django's distinction between BaseIterable and ModelIterable, I think our SpecificIterable returns a list of model instances and so is a ModelIterable at heart.